### PR TITLE
Add tests for file metrics for Kotlin

### PR DESCRIPTION
**Pull request by michaelalper-cezik-mw**
Adds test cases and source files and for Kotlin programming language. Source files were tested to compile correctly.

### CODE REVIEW

File: test/parser/KotlinMetrics.test.ts [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+
+        it("should count case but no default statements correctly", async () => {
+            await testFileMetric(
+                kotlinTestResourcesPath + "case-statements.kt",
+                FileMetric.complexity,
+                5
+            );
+        });
+
+        it("should count functions and methods correctly", async () => {
```
Comment (mylinhdao) [inline test/parser/KotlinMetrics.test.ts:32]: Looks like init block is counted for complexity and functions metric for Kotlin, but not Java. Maybe we should mention this in our README.md? @ResistantBear @michaelalper-cezik-mw 
```suggestion
        it("should count all function declarations and one init block correctly", async () => {
```

File: test/parser/KotlinMetrics.test.ts [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+            await testFileMetric(
+                kotlinTestResourcesPath + "functions-and-methods.kt",
+                FileMetric.complexity,
+                11
+            );
+        });
+
+        it("should not count multiple return statements within functions and methods correctly", async () => {
```
Comment (mylinhdao) [inline test/parser/KotlinMetrics.test.ts:40]:
```suggestion
        it("should not count multiple return statements within functions and methods", async () => {
```

**Review by mylinhdao — COMMENTED** [review]
In general LGTM, but the remarks must be discussed.

File: test/parser/KotlinMetrics.test.ts [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+                FileMetric.complexity,
+                10
+            );
+        });
+
+        it("should not count any class declaration", async () => {
+            await testFileMetric(kotlinTestResourcesPath + "classes.kt", FileMetric.complexity, 4);
+        });
+
+        it("should count case but no default statements correctly", async () => {
```
[the remark this answers is not in the document]
Comment (michaelalper-cezik-mw) [reply test/parser/KotlinMetrics.test.ts:24]: @ce-bo suggested we should not skip any tests (can't find the post).
After #64 is implemented these tests will fail and we can correct the expected complexity.

File: test/parser/KotlinMetrics.test.ts (hunk 1 above)
Comment (ResistantBear) [reply test/parser/KotlinMetrics.test.ts:32]: I guess the init-block is a block where you can throw arbitrary statements in, just like a function body? In this case, we should count it as a function, just like contructors, etc.

File: test/parser/KotlinMetrics.test.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (ce-bo) [reply test/parser/KotlinMetrics.test.ts:24]: To wait for #64 blocks this PR from being merged. I think you mean the same: I suggest to change the test name to also count default labels for now, as this is the current behavior of metric-gardener and update it in #64 later when the test fails.

File: test/parser/KotlinMetrics.test.ts (hunk 1 above)
Comment (ce-bo) [reply test/parser/KotlinMetrics.test.ts:32]: I agree that this is like a function and should be counted accordingly for complexity and functions metric.
Do we handle init blocks for java?

**Comment by ce-bo** [comment]
I am super happy that you add such extensive tests with example files 🥳

File: test/parser/KotlinMetrics.test.ts (hunk 1 above)
Comment (mylinhdao) [reply test/parser/KotlinMetrics.test.ts:32]: @ce-bo actually, init block are not counted in Java. But we will change that in #108

**Review by mylinhdao — APPROVED** [review]
all remarks are solved

**Pull request merged**